orchestrationSpecs/userConfigSchemas.txt

```diff
@@ -0,0 +1,59 @@
+HTTP_AUTH_BASIC: {
+  username: string
+  password: string
+}
+
+HTTP_AUTH_SIGV4: {
+  region:  string
+  service?: string | "es" // default
+}
+
+CLUSTER_CONFIG: {
+  name:            string
+  endpoint?:       string // optional for source and maybe metadata evaluate commands, but required for most others
+  allow_insecure?: bool
+  version?:        string // required in some contexts?
```

Comment:
This should be typed - use/reimplement `transformation/src/main/java/org/opensearch/migrations/Version.java`